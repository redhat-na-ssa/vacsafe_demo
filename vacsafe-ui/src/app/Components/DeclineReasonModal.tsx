import * as React from 'react';
import { Button, Modal, ModalVariant, Radio, Stack } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';
import { useState } from 'react';

const DeclineReasonModal = ({...props}) => {
  const [checked, setChecked] = useState(""); 
  const declineReasons = [
    "Attachments not legible",
    "Supporting document missing vaccine information",
    "Supporting document missing name and/or DOB",
    "Not a valid form of supporting documentation",
    "Supporting documentation mismatch with employee information",
    "Supporting documentation provided is a copy of the wrong side of the card"
  ]

  const handleRadioChange = (_, event) => {
    props.setRejectReason(event.currentTarget.value);
    setChecked(event.currentTarget.name);
  }

  return (
    <Modal
      actions={[
        <Button
          icon={<CheckCircleIcon />}
          isDisabled={props.isSendingReview}
          isLoading={props.isSendingReview}
          key="confirm"
          variant="primary"
          onClick={() => props.sendReview(false)}
        >
          Confirm
        </Button>,
        <Button
          key="cancel"
          onClick={() => props.setIsModalOpen(false)}
          variant="plain"
        >
          Cancel
        </Button>
      ]}
      isOpen={props.isModalOpen}
      onClose={() => props.setIsModalOpen(false)}
      title="Reason For Decline - Select Most Appropriate"
      variant={ModalVariant.medium}
    >
      <Stack hasGutter>
        {declineReasons.map((reason, index) => (
          <Radio
            key={"radio-" + index}
            isChecked={checked === "radio-" + index}
            name={"radio-" + index}
            onChange={handleRadioChange}
            label={reason}
            id={"radio-controlled-" + index}
            value={reason}
          />
        ))}
      </Stack>
    </Modal>
  )
}

export default DeclineReasonModal;