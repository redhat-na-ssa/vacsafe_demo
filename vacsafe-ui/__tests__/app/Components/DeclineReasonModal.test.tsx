import React from 'react';
import { shallow } from 'enzyme';
import DeclineReasonModal from '../../../src/app/Components/DeclineReasonModal';

describe('<DeclineReasonModal />', ()=> {
    it('should be defined', () => {
        expect(DeclineReasonModal).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<DeclineReasonModal />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<DeclineReasonModal />)
        expect(wrapper).toMatchSnapshot()
    })
})
