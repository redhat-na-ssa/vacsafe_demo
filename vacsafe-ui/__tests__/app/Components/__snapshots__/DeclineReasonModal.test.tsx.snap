// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeclineReasonModal /> should match existing snapshot 1`] = `
<Modal
  actions={
    Array [
      <Button
        icon={
          <CheckCircleIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        }
        onClick={[Function]}
        variant="primary"
      >
        Confirm
      </Button>,
      <Button
        onClick={[Function]}
        variant="plain"
      >
        Cancel
      </Button>,
    ]
  }
  appendTo={[Function]}
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isOpen={false}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Reason For Decline - Select Most Appropriate"
  titleIconVariant={null}
  titleLabel=""
  variant="medium"
>
  <Stack
    hasGutter={true}
  >
    <Radio
      className=""
      id="radio-controlled-0"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      key="radio-0"
      label="Attachments not legible"
      name="radio-0"
      onChange={[Function]}
      value="Attachments not legible"
    />
    <Radio
      className=""
      id="radio-controlled-1"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      key="radio-1"
      label="Supporting document missing vaccine information"
      name="radio-1"
      onChange={[Function]}
      value="Supporting document missing vaccine information"
    />
    <Radio
      className=""
      id="radio-controlled-2"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      key="radio-2"
      label="Supporting document missing name and/or DOB"
      name="radio-2"
      onChange={[Function]}
      value="Supporting document missing name and/or DOB"
    />
    <Radio
      className=""
      id="radio-controlled-3"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      key="radio-3"
      label="Not a valid form of supporting documentation"
      name="radio-3"
      onChange={[Function]}
      value="Not a valid form of supporting documentation"
    />
    <Radio
      className=""
      id="radio-controlled-4"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      key="radio-4"
      label="Supporting documentation mismatch with employee information"
      name="radio-4"
      onChange={[Function]}
      value="Supporting documentation mismatch with employee information"
    />
    <Radio
      className=""
      id="radio-controlled-5"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      key="radio-5"
      label="Supporting documentation provided is a copy of the wrong side of the card"
      name="radio-5"
      onChange={[Function]}
      value="Supporting documentation provided is a copy of the wrong side of the card"
    />
  </Stack>
</Modal>
`;
